README.md

```diff
@@ -68,23 +70,32 @@ or
 yarn add @guardian/cdk
 ```
 
-Patterns can be imported from the top level of the library (e.g. `import { InstanceRole } from "@guardian/cdk";`) while constructs must be imported from their construct directory (e.g. `import { GuAutoScalingGroup } from "@guardian/cdk/lib/constructs/autoscaling";`)
+Patterns can be imported from the top level of the library:
 
-There are more details on using the CDK library in [docs](docs)
+```typescript
+import { GuScheduledLambda } from "@guardian/cdk";
+```
 
-## Releasing
+While constructs must be imported from their construct directory:
 
-We use [`np`](https://www.npmjs.com/package/np) to help orchestrate the release process.
-To release a new version, run `./script/release`. You will need to be logged in to your `npm` account (`npm login`) which must be part of the Guardian organisation. If you have 2fa enabled, you will be prompted for an OTP during the release process.
+```typescript
+import { GuAutoScalingGroup } from "@guardian/cdk/lib/constructs/autoscaling";
+```
 
-The script automatically prepares a [GitHub Release](https://docs.github.com/en/github/administering-a-repository/about-releases) and opens a pre-populated form in your browser. Please complete the GitHub Release creation manually.
+This is intentional as the patterns ideally solve the majority of use-cases.
+if they don't, PRs are welcome!
 
-It will also automatically create a PR to edit `version` in [`package.json`](./package.json).
+There are more details on using the CDK library in [docs](docs)
 
-### A note about release numbers
+## Releasing
 
-This library follows [semantic versioning](https://semver.org/). Part of the semantic versioning specification [states](https://semver.org/#spec-item-4):
+TL;DR We release new versions of the library to NPM automagically.
 
-> Major version zero (0.y.z) is for initial development. Anything MAY change at any time. The public API SHOULD NOT be considered stable.
+We use [semantic-release](https://github.com/semantic-release/semantic-release) and [guardian/post-release-action](https://github.com/guardian/post-release-action) to automate releases.
 
-For simplicity, this library only has minor version releases whilst still on a 0 major number.
+To release a new version:
+1. Raise a PR following using [Angular](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commits) / [Karma](http://karma-runner.github.io/6.1/dev/git-commit-msg.html) commits
+1. Get your PR reviewed and approved
+1. Merge your PR
+1. Wait... The robots will parse your commits to determine the next version number (following [semantic versioning](https://semver.org/)), release to npm and update `package.json`. Automatically.
+1. Enjoy a comment on your PR to inform you that your change has been released.
```

Comment:
AFAIK PR titles can be used if you squash and merge as that's the commit you get on the main branch. Alternatively, you can ensure the commits conform and then merge as usual. At the moment the PR title is validated in CI and the commits don't have to conform but if you were to merge (rather than squash + merge) with a conforming PR and non-conforming commits, you wouldn't trigger a release.

I don't know if you want to enforce one style over the other or allow both. If it's both, maybe this should be a bit clearer on the two options and if it's one over the other, then the checks may want to change accordingly?

Reply:
Ah, great point. Will discuss merge styles with team next week and update accordingly.

Reply:
I think you can [configure the repo so that you can _only_ use squash and merge](https://docs.github.com/en/github/administering-a-repository/configuring-commit-squashing-for-pull-requests)? This would probably get my vote as it means we are only enforcing conventions where it is strictly necessary. It also means that if someone creates a load of commit messages which don't conform, they can easily fix things by updating their PR title and hitting merge, which reduces friction for contributors.

Reply:
Happy to change this setting. IIUC (for all intents and purposes) the only impact is the merge button on a PR is worded differently - there is no impact to our git workflow otherwise.

@stephengeller any objections to changing the repo settings to only allow squash and merge?

Reply:
Have changed the repo settings now to not allow merge commits; I think this resolves this thread?

Reply:
Maybe the first point should be more explicit that it's only the PR title which needs to conform to the style. And similar with the penultimate point about `parse you commits`?

```
1. Raise a PR with a title that conforms to [Angular](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commits) / [Karma](http://karma-runner.github.io/6.1/dev/git-commit-msg.html) standards
...
1. Wait... The robots will parse your merge commit to determine the next version number (following [semantic versioning](https://semver.org/)), release to npm and update `package.json`. Automatically.
```

Reply:
Have reworded the entire section as it was pretty verbose.